### PR TITLE
Matrix rust static analysis job

### PR DESCRIPTION
**Pull request by jamilbk**
Addresses https://github.com/firezone/firezone/pull/2293#discussion_r1353461468

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -8,8 +8,23 @@ defaults:
...
+      matrix:
+        runs-on:
+          # We only need to run static analysis checks per OS family
+          - ubuntu-latest
+          - macos-13
+          - windows-2022
+        # TODO: https://github.com/rust-lang/cargo/issues/5220
+        include:
+          - runs-on: ubuntu-latest
+            packages: -p firezone-headless-client -p firezone-gateway -p connlib-client-android
```
Comment (thomaseizinger) [inline .github/workflows/rust.yml]: Can we just leave this empty? I run clippy on all rust packages all the time locally and it passes.

Comment (conectado) [reply .github/workflows/rust.yml]: I think in ubuntu yeah but in windows there are problems.
Being the case that we just can't compile everything in all platforms I just preferred being explicit in a per-platform basis, this way we are explicit on what package should run on what platform *always*

Comment (thomaseizinger) [reply .github/workflows/rust.yml]: Yeah I was specifically talking about the ubuntu case! I'd just like to have at least 1 CI job that lints all packages without having to manually list them. As we create new crates, this is likely to not get updated.

Comment (jamilbk) [reply .github/workflows/rust.yml]: Updated to lint all packages with a comment explaining why.

**Review by AndrewDryga — APPROVED** [review]

**Pull request merged**